master/internal/command/command_intg_test.go

```diff
@@ -7,42 +7,239 @@ import (
+	require.Equal(t, len(resp2.Notebooks), 2)
+
+	// Kill 1 Notebook.
+	resp3, err := DefaultCmdService.KillNTSC(cmd2.Id, model.TaskTypeNotebook)
+	require.NotNil(t, resp3)
+	require.NoError(t, err)
+
+	nb3 := resp3.ToV1Notebook()
+	require.Equal(t, nb3.State, taskv1.State_STATE_TERMINATED)
+
+	// Set Notebook priority.
+	resp4, err := DefaultCmdService.SetNTSCPriority(cmd1.Id, 0, model.TaskTypeNotebook)
+	require.NotNil(t, resp4)
+	require.NoError(t, err)
+}
+
+func TestShellManagerLifecycle(t *testing.T) {
+	db := setupTest(t)
+	ctx := context.TODO()
+
+	// Launch a Shell.
+	cmd1 := launchShell(ctx, t, db)
+
+	// Get Shell.
+	resp1, err := DefaultCmdService.GetShell(&apiv1.GetShellRequest{ShellId: cmd1.Id})
+	require.NotNil(t, resp1)
+	require.NoError(t, err)
+
+	// Launch another Shell.
+	cmd2 := launchShell(ctx, t, db)
+
+	// Get Shells.
+	resp2, err := DefaultCmdService.GetShells(&apiv1.GetShellsRequest{})
+	require.NotNil(t, resp2)
+	require.NoError(t, err)
+	require.Equal(t, len(resp2.Shells), 2)
+
+	// Kill 1 Shell.
+	resp3, err := DefaultCmdService.KillNTSC(cmd2.Id, model.TaskTypeShell)
+	require.NotNil(t, resp3)
+	require.NoError(t, err)
+
+	shell3 := resp3.ToV1Shell()
+	require.Equal(t, shell3.State, taskv1.State_STATE_TERMINATED)
+
+	// Set Shell priority.
+	resp4, err := DefaultCmdService.SetNTSCPriority(cmd1.Id, 0, model.TaskTypeShell)
+	require.NotNil(t, resp4)
+	require.NoError(t, err)
+}
+
+func TestTensorboardManagerLifecycle(t *testing.T) {
+	db := setupTest(t)
+	ctx := context.TODO()
+
+	// Launch a Tensorboard.
+	cmd1 := launchTensorboard(ctx, t, db)
+
+	// Get Tensorboard.
+	resp1, err := DefaultCmdService.GetTensorboard(&apiv1.GetTensorboardRequest{TensorboardId: cmd1.Id})
+	require.NotNil(t, resp1)
+	require.NoError(t, err)
+
+	// Launch another Tensorboard.
+	cmd2 := launchTensorboard(ctx, t, db)
+
+	// Get Tensorboards.
+	resp2, err := DefaultCmdService.GetTensorboards(&apiv1.GetTensorboardsRequest{})
+	require.NotNil(t, resp2)
+	require.NoError(t, err)
+	require.Equal(t, len(resp2.Tensorboards), 2)
+
+	// Kill 1 Tensorboard.
+	resp3, err := DefaultCmdService.KillNTSC(cmd2.Id, model.TaskTypeTensorboard)
+	require.NotNil(t, resp3)
+	require.NoError(t, err)
+
+	tb3 := resp3.ToV1Tensorboard()
+	require.Equal(t, tb3.State, taskv1.State_STATE_TERMINATED)
+
+	// Set Tensorboard priority.
+	resp4, err := DefaultCmdService.SetNTSCPriority(cmd1.Id, 0, model.TaskTypeTensorboard)
+	require.NotNil(t, resp4)
+	require.NoError(t, err)
+}
+
+func setupTest(t *testing.T) *db.PgDB {
 	pgDB := db.MustSetupTestPostgres(t)
-	task := db.RequireMockTask(t, pgDB, nil)
+	// First init the new Command Service
+	var mockRM mocks.ResourceManager
+	sub := sproto.NewAllocationSubscription(queue.New[sproto.ResourcesEvent](), func() {})
+	mockRM.On("Allocate", mock.Anything, mock.Anything).Return(sub, nil)
+	mockRM.On("Release", mock.Anything, mock.Anything).Return(nil)
+	mockRM.On("SetGroupPriority", mock.Anything, mock.Anything).Return(nil)
```

Comment:
this is in enough places we could stand to have a dedicated "mockRMThatAlwaysJustWorks" helper.

Reply:
worth another ticket, but imo not here

Reply:
https://hpe-aiatscale.atlassian.net/browse/DET-9967